### PR TITLE
Autocomplete css custom property names.

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-editor-service/101)

### CODE REVIEW

File: src/test/auto-completer_test.ts [hunk 1]
```diff
@@ -474,6 +474,79 @@ suite('AutoCompleter', () => {
               'polymer/element-with-databinding.html', internalPropUsePosition),
           databindingCompletions);
     });
+
+
+    test(`autocomplete css custom properties`, async() => {
```
Comment (valdrinkoshi) [inline src/test/auto-completer_test.ts]: nit: can you separate this in 2 tests - 1 for declaration and 1 for completion?

File: src/test/auto-completer_test.ts [hunk 2]
```diff
@@ -474,6 +474,79 @@ suite('AutoCompleter', () => {
...
+      await client.openFile('index.html', `
+        <style>
+          div {
+            --foo: red;
+            --bar: var(--baz);
+            --fizz: {--floo: var(--flub);};
+          }
+        </style>
+      `);
+      const declaractionPosition = {line: 3, column: 16};
```
Comment (valdrinkoshi) [inline src/test/auto-completer_test.ts]: Is this correct? In vscode this spot corresponds to the location right after `div`
screen shot 2017-12-06 at 10 12 05 am

File: src/test/auto-completer_test.ts [hunk 3]
```diff
@@ -474,6 +474,79 @@ suite('AutoCompleter', () => {
...
+      const usePosition = {line: 4, column: 27};
```
Comment (valdrinkoshi) [inline src/test/auto-completer_test.ts]: Similar to above, in vscode this location does not exist, the column goes up to 26 
screen shot 2017-12-06 at 10 12 22 am

File: src/language-server/auto-completer.ts [hunk 4]
```diff
@@ -109,6 +117,53 @@ export default class AutoCompleter extends Handler {
...
+  }
+
+  private getCustomPropertyUseCompletions(document: Document) {
+    const propertyAssignments = document.getFeatures({
+      kind: 'css-custom-property-assignment',
+      imported: true,
+      externalPackages: true
+    });
+    const names = new Set<string>([...propertyAssignments].map(a => a.name));
+    const items = [...names].sort().map((name): CompletionItem => {
```
[the remark this answers is not in the document]
Comment (rictic) [reply src/language-server/auto-completer.ts]: We need the map to make the set of names, to eliminate duplicates.

File: src/test/auto-completer_test.ts (hunk 1 above)
Comment (rictic) [reply src/test/auto-completer_test.ts]: Sure, done

File: src/test/auto-completer_test.ts (hunk 2 above)
Comment (rictic) [reply src/test/auto-completer_test.ts]: Tools report 1-indexed, code uses 0-indexed

File: src/test/auto-completer_test.ts (hunk 3 above)
Comment (rictic) [reply src/test/auto-completer_test.ts]: Tools report 1-indexed, code uses 0-indexed

**Pull request merged**